client/src/views/Search.vue

```diff
@@ -37,11 +37,12 @@ export default defineComponent({
     },
     methods: {
         async onAddKeyword(event: { value: string }) {
-            this.searchTerms.push(event.value);
+            this.searchTerms = this.searchTerms.concat(event.value);
```

Comment:
@micahh2 This change is again not really necessary or even worse because you are ditching the current array and assigning a new one.

Reply:
As mentioned above, in general destructive operators (such as push and slice) aren't good practice because they often have unintended side effects. Functions and components are often easier to reason about when state changes must be created from new state. There is, of course, exceptions to this, but I'm just trying to keep things consistent.